Source/Core/DolphinQt2/FIFO/FIFOAnalyzer.cpp

```diff
@@ -0,0 +1,500 @@
+        new_label = QStringLiteral("NOP");
+        break;
+
+      case 0x44:
+        new_label = QStringLiteral("0x44");
+        break;
+
+      case OpcodeDecoder::GX_CMD_INVL_VC:
+        new_label = QStringLiteral("GX_CMD_INVL_VC");
+        break;
+
+      case OpcodeDecoder::GX_LOAD_CP_REG:
+      {
+        u32 cmd2 = *objectdata++;
+        u32 value = Common::swap32(objectdata);
+        objectdata += 4;
+
+        new_label = QStringLiteral("CP  %1  %2")
+                        .arg(cmd2, 2, 16, QLatin1Char('0'))
+                        .arg(value, 8, 16, QLatin1Char('0'));
+      }
+      break;
+
+      case OpcodeDecoder::GX_LOAD_XF_REG:
+      {
+        u32 cmd2 = Common::swap32(objectdata);
+        objectdata += 4;
+
+        u8 streamSize = ((cmd2 >> 16) & 15) + 1;
+
+        const u8* stream_start = objectdata;
+        const u8* stream_end = stream_start + streamSize * 4;
+
+        new_label = QStringLiteral("XF  %1  ").arg(cmd2, 16, 8, QLatin1Char('0'));
+        while (objectdata < stream_end)
+        {
+          new_label += QStringLiteral("%1").arg(*objectdata++, 16, 2, QLatin1Char('0'));
+
+          if (((objectdata - stream_start) % 4) == 0)
+            new_label += QLatin1Char(' ');
+        }
+      }
+      break;
+
+      case OpcodeDecoder::GX_LOAD_INDX_A:
+      case OpcodeDecoder::GX_LOAD_INDX_B:
+      case OpcodeDecoder::GX_LOAD_INDX_C:
+      case OpcodeDecoder::GX_LOAD_INDX_D:
+      {
+        objectdata += 4;
+        new_label = QStringLiteral("LOAD INDX %1")
+                        .arg((command == OpcodeDecoder::GX_LOAD_INDX_A) ?
+                                 QStringLiteral("A") :
+                                 (command == OpcodeDecoder::GX_LOAD_INDX_B) ?
+                                 QStringLiteral("B") :
+                                 (command == OpcodeDecoder::GX_LOAD_INDX_C) ? QStringLiteral("C") :
+                                                                              QStringLiteral("D"));
+      }
+      break;
+
+      case OpcodeDecoder::GX_CMD_CALL_DL:
+        // The recorder should have expanded display lists into the fifo stream and skipped the
+        // call to start them
+        // That is done to make it easier to track where memory is updated
+        ASSERT(false);
+        objectdata += 8;
+        new_label = QStringLiteral("CALL DL");
+        break;
+
+      case OpcodeDecoder::GX_LOAD_BP_REG:
+      {
+        u32 cmd2 = Common::swap32(objectdata);
+        objectdata += 4;
+        new_label = QStringLiteral("BP  %02X %06X")
+                        .arg(cmd2 >> 24, 2, 16, QLatin1Char('0'))
+                        .arg(cmd2 & 0xFFFFFF, 6, 16, QLatin1Char('0'));
+      }
+      break;
+
+      default:
+        new_label = tr("Unexpected 0x80 call? Aborting...");
+        objectdata = static_cast<const u8*>(next_objdata_start);
+        break;
+      }
+      new_label = QStringLiteral("%1:  ").arg(new_offset, 8, 16, QLatin1Char('0')) + new_label;
```

Comment:
Are those two spaces intentional (for alignment or whatever)?

Reply:
Yep.